pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -776,6 +752,235 @@ func TestUpdateStatus(t *testing.T) {
 	})
 }
 
+func TestDisabledUserAccount(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+	username := "johndoe"
+	userID := uuid.NewV4().String()
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+
+	userAcc := newUserAccount(username, userID)
+	userUID := types.UID(username + "user")
+	preexistingIdentity := &userv1.Identity{ObjectMeta: metav1.ObjectMeta{
+		Name:      ToIdentityName(userAcc.Spec.UserID),
+		Namespace: "toolchain-member",
+		UID:       types.UID(username + "identity"),
+	}, User: corev1.ObjectReference{
+		Name: username,
+		UID:  userUID,
+	}}
+	preexistingUser := &userv1.User{ObjectMeta: metav1.ObjectMeta{
+		Name:            username,
+		Namespace:       "toolchain-member",
+		UID:             userUID,
+		OwnerReferences: []metav1.OwnerReference{},
+	}, Identities: []string{ToIdentityName(userAcc.Spec.UserID)}}
+	preexistingNsTmplSet := &toolchainv1alpha1.NSTemplateSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      userAcc.Name,
+			Namespace: "toolchain-member",
+		},
+		Spec: newNSTmplSetSpec(),
+		Status: toolchainv1alpha1.NSTemplateSetStatus{
+			Conditions: []toolchainv1alpha1.Condition{
+				{Type: toolchainv1alpha1.ConditionReady, Status: corev1.ConditionTrue},
+			},
+		},
+	}
+
+	t.Run("disabling useraccount", func(t *testing.T) {
+		// given
+		r, req, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity, preexistingNsTmplSet)
+
+		//when
+		res, err := r.Reconcile(req)
```

Comment:
What do you need this reconcile for?

Reply:
this reconcile enters us into this condition: if !util.IsBeingDeleted(userAcc) && !userAcc.Spec.Disabled {...} prior to setting disabled to true

Reply:
This condition should be setup as the pre-configuration in prepareReconcile(), shouldn't it?

Reply:
@alexeykazakov yeah, that's right! forgot! sorry, fixed!